indigo/indigo/src/main/scala/indigo/shared/scenegraph/Graphic.scala

```diff
@@ -98,10 +104,19 @@ final case class Graphic[M <: Material](
   def toShaderData: ShaderData =
     material.toShaderData
 
+  def onEvent(e: ((Rectangle, GlobalEvent)) => List[GlobalEvent]): Graphic[M] =
+    this.copy(eventHandler = e, eventHandlerEnabled = true)
+  def enableEvents: Graphic[M] =
+    this.copy(eventHandlerEnabled = true)
```

Comment:
Yes, in this version. My expectation is that most people will just add a handler, which will enabled the handler, and that will be the end of it for them. disable and enable I image will be the exceptional usage rather than the norm.

It does feel clunky though so I'm still thinking about it - just getting something working at the moment. More familiar semantics would be adding and removing listeners (or things that looked like listeners) but that means even more complications to the primitive scene node types.